app/pages/graph.jsx

```diff
@@ -0,0 +1,168 @@
+
+  nodes.forEach((node) => {
+    dagreGraph.setNode(node.id, { width: nodeWidth, height: nodeHeight })
+  })
+
+  edges.forEach((edge) => {
+    dagreGraph.setEdge(edge.source, edge.target)
+  })
+
+  dagre.layout(dagreGraph)
+
+  nodes.forEach((node) => {
+    const nodeWithPosition = dagreGraph.node(node.id)
+    node.targetPosition = isHorizontal ? "left" : "top"
+    node.sourcePosition = isHorizontal ? "right" : "bottom"
+
+    // We are shifting the dagre node position (anchor=center center) to the top left
+    // so it matches the React Flow node anchor point (top left).
+    node.position = {
+      x: nodeWithPosition.x - nodeWidth / 2,
+      y: nodeWithPosition.y - nodeHeight / 2,
+    }
+
+    return node
+  })
+
+  return { nodes, edges }
+}
+
+const Graph = () => {
+  const currentUser = useCurrentUser()
+  const session = useSession()
+  const currentWorkspace = useCurrentWorkspace()
+  const router = useRouter()
+  const [drafts, { refetch }] = useQuery(getDrafts, { session })
+  const [{ nodesData: nodesQuery, edgesData: edgesQuery }] = useQuery(getNodes, undefined)
+  const [invitations] = useQuery(getInvitedModules, { session })
+  const [nodes, setNodes, onNodesChange] = useNodesState([])
+  const [edges, setEdges, onEdgesChange] = useEdgesState([])
+  const [onlyConnected, setOnlyConnected] = useState(true)
+  const prefersDarkMode = useMediaPredicate("(prefers-color-scheme: dark)")
+
+  useEffect(() => {
+    edgesQuery.forEach((edge) => {
+      edge.style = { stroke: prefersDarkMode ? "#fff" : "#000" }
+      return edge
+    })
+
+    if (onlyConnected) {
+      const x = nodesQuery.filter((node) => {
+        return edgesQuery.some((edge) => {
+          return edge.source === node.id || edge.target === node.id
+        })
+      })
+      const ele = getLayoutedElements(x, edgesQuery, "TB")
+      setNodes(ele.nodes)
+      setEdges(ele.edges)
+    } else {
+      const ele = getLayoutedElements(nodesQuery, edgesQuery, "TB")
+      setNodes(ele.nodes)
+      setEdges(ele.edges)
+    }
+  }, [onlyConnected, nodesQuery, edgesQuery, prefersDarkMode, setEdges, setNodes])
+
+  return (
+    <>
+      <Navbar
+        currentUser={currentUser}
+        session={session}
+        currentWorkspace={currentWorkspace}
+        router={router}
+        drafts={drafts}
+        invitations={invitations}
+        refetchFn={refetch}
+      />
+      <div className="h-[85vh] w-full">
+        <ReactFlow
+          nodes={nodes}
+          edges={edges}
+          panOnScroll={true}
+          nodeTypes={nodeTypes}
+          minZoom={-0.3}
+          onNodesChange={onNodesChange}
+          onEdgesChange={onEdgesChange}
+          fitView
+        >
+          <MiniMap
+            style={{ backgroundColor: prefersDarkMode ? "#1e293b" : "#fff", color: "red" }}
+            nodeColor={prefersDarkMode ? "#fff" : "#000"}
+          />
+          <Controls style={{ fill: prefersDarkMode ? "#fff" : "#000" }}>
```

Comment:
@chartgerink  I saw that you added a margin in the commit, https://github.com/libscie/ResearchEquals.com/pull/378/commits/d49c7defd45beade6e424720bb2d217886eef367. Resolving this conversation.

Reply:
Thanks @nsunami - sorry for not responding here sooner. 

This is a great catch - I didn't have chat enabled on my test instance so it slipped my radar! I decided to go with your suggestion, as this would require some more messing about in `Layout` to do it route dependent (which to be honest  I tried real quick but couldn't make reliable in a way I wanted).